plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcDeleteDeltaPageSource.java

```diff
@@ -76,20 +76,20 @@
     private boolean closed;
 
     public static Optional<ConnectorPageSource> createOrcDeleteDeltaPageSource(
-            Path path,
+            String path,
             long fileSize,
             OrcReaderOptions options,
             ConnectorIdentity identity,
-            Configuration configuration,
             HdfsEnvironment hdfsEnvironment,
-            FileFormatDataSourceStats stats)
+            FileFormatDataSourceStats stats,
+            TrinoFileSystemFactory trinoFileSystemFactory)
     {
         OrcDataSource orcDataSource;
         try {
-            FileSystem fileSystem = hdfsEnvironment.getFileSystem(identity, path, configuration);
-            FSDataInputStream inputStream = hdfsEnvironment.doAs(identity, () -> fileSystem.open(path));
+            TrinoFileSystem fileSystem = trinoFileSystemFactory.create(identity);
+            TrinoInput inputStream = hdfsEnvironment.doAs(identity, () -> fileSystem.newInputFile(path).newInput());
```

Comment:
you don't need to wrap in `doAs` anymore. See `io.trino.filesystem.hdfs.HdfsInputFile#newInput`